registration/include/pcl/registration/impl/ndt_2d.hpp

```diff
@@ -336,8 +336,8 @@ namespace pcl
         test (const PointT& transformed_pt, const double& cos_theta, const double& sin_theta) const
         {
           ValueAndDerivatives<3,double> r = ValueAndDerivatives<3,double>::Zero ();
-          for (size_t i = 0; i < 4; i++)
-              r += single_grids_[i]->test (transformed_pt, cos_theta, sin_theta);
+          for (auto &single_grid : single_grids_)
```

Comment:
I think it should be possible to make it `const` (if I traced `test` method correctly it is marked `const`).